sliceutil/sets.go

```diff
@@ -57,3 +57,21 @@ func Intersection[T comparable](a, b []T) []T {
 	}
 	return inter
 }
+
+// IntersectionOfMany returns the elements common to all arguments
+func IntersectionOfMany[T comparable](slices ...[]T) []T {
+	if len(slices) == 1 {
+		return slices[0]
+	}
```

Comment:
This is probably not necessary as you have `common = slices[0]` below and with `continue` on first item, it should return immediately after that, albeit it's not bad to have this here 🤔 

On the other hand it might be worth checking `if len(slices) == 0` and then `return nil` as otherwise the line below would panic 😅 Edge case, but if you have some `sliceOfSlices = foo()` followed by `IntersectionOfMany(sliceOfSlices...)` and the result was empty, you'd get an out of bounds panic

Reply:
@ripexz, great caught!!!
i've rearranged a little bit, now if there are is no input we will return empty slice, this might be easier to handle when using `IntersectionOfMany()` as it will always return same type. We got this covered in tests with `TestIntersectionOfMany():EmptyLists` example